src/controllers/controller.line.js

```diff
@@ -111,11 +111,14 @@ LineController.defaults = {
   datasetElementType: 'line',
   dataElementType: 'point',
 
-  datasets: {
-    showLine: true,
-    spanGaps: false,
-  },
+  showLine: true,
+  spanGaps: false,
+};
 
+/**
+ * @type {any}
+ */
+LineController.overrides = {
```

Comment:
It's clearer now which options apply to the dataset and which ones apply to the chart. For example, it was confusing why `scales` and `plugins` was mixed with the other options at the dataset level under `defaults.controllers.line`.